tiny86/src/Syscall/Internal.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Syscall.Internal
+    ( SyscallState(..)
+    , SyscallStateReg
+    , Syscall(..)
+    , SyscallReg
+    , SyscallDFA
+    , SyscallDFAState(..)
+    , stepDataBytes
+    , stepPtrBytes
+    ) where
+
+import Clash.Prelude
+
+stepDataBytes :: Int
+stepDataBytes = 2 * 4 -- the number of data bytes present per step.
+
+stepPtrBytes :: Int
+stepPtrBytes = 2 * 4 -- the number of address bytes present per step.
+
+-- see: https://cgc-docs.legitbs.net/libcgc/cgcabi/
+data Syscall
+    = SYSCALL_TERMINATE
+    | SYSCALL_TRANSMIT
+    | SYSCALL_RECIEVE
+    | SYSCALL_FDWAIT
+    | SYSCALL_ALLOCATE
+    | SYSCALL_DEALLOCATE
+    | SYSCALL_RANDOM
+    deriving (Bounded, Show)
+
+type SyscallReg = BitVector 32
+
+-- auto-derived Enum on Syscall (reasonably) enumerates from 0;
+-- so, implement Enum instance manually.
+instance Enum Syscall where
```

Comment:
Again, in future, perhaps we should revisit the choice of using the DECREE syscall model here, but should be fine for now.